feat(model+query): support `options` parameter for distinct()

vkarpov15 opened the pull request:
Fix #8006

<!-- Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. The two fields below are mandatory.

If you're making a change to documentation, do **not** modify a `.html` file directly. Instead, find the corresponding `.pug` file or test case in the `test/docs` directory. -->

**Summary**

Make it easier to use `distinct()` with transactions, so you don't need to chain `setOptions()`. `distinct()` looks to be the only query function where you can't pass in `options`.

<!-- Explain the **motivation** for making this change. What problem does the pull request solve? -->

**Examples**

<!-- If this code fixes a bug or adds a new feature, provide an example demonstrating the change, unless you added a test. -->

hasezoey commented on test/docs/transactions.test.js:
```diff
@@ -338,6 +338,26 @@ describe('transactions', function() {
   });
 
+  it('distinct (gh-8006)', async function() {
+    const Character = db.model('gh8006_Character', new Schema({ name: String, rank: String }, { versionKey: false }));
+
+
+    const session = await db.startSession();
```
```suggestion
    const Character = db.model('gh8006_Character', new Schema({ name: String, rank: String }, { versionKey: false }));

    const session = await db.startSession();
```

i dont know if this intended, but i think a single new line would be enough

hasezoey reviewed (changes requested):
LGTM, but the `options` parameter still needs to be added to the query types (currently only in the model's types) and a similar test to `using options parameter for distinct` on query should likely be in for Model too, or is the transaction test enough?

As a note, the `Model.distinct` and `Query.prototype.distinct` have differing `options` checks, is this intentional?

vkarpov15 commented:
Thanks for the suggestions @hasezoey , I've implemented them both. Added `options` param to query `distinct()` types and removed the `typeof options === 'object'` check in Query.prototype.distinct (and countDocuments() + estimatedDocumentCount() for consistency)
